### ISSUE TITLE
Questions about AzureLogAnalycsSink.PostDataAsync()

### ISSUE THREAD

**Issue by Morcatko** [saleem-mirza/serilog-sinks-azure-analytics#22]
Hello,
Thanks for this sink, It was a matter of seconds to set it up and get some logs (well seconds to set it up, and then minutes wating for Azure to index it :) )

I have few questions about the code.
1) In `AzureLogAnalycsSink.PostDataAsync(...)` you have
```
var response = client.PostAsync(_analyticsUrl, stringContent).Result;
```
Is there any reason why this uses `.Result` rather than `await`? I believe this call might take some time as it   create the network connection (DNS lookup, TCP connection, SSL handshake, ...)

2) The same line as in 1)
`response` is of `HttpResponseMessage` which is `IDisposable` but it is not diposed anywhere

3) You are creating `new HttpClient()` for each batch (which is every 10s by default AFAIK)
There might be some issues with this approach, which are hard to debug and reproduce. I recommend reading this - https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/
Long story short - make your HttpClient static and do not dispose it.

**Comment by saleem-mirza** [comment]
@Morcatko, thank you for critical review. I really appreciate it.

1) not using await is probably a mistake. It got overlooked when converting from synchronous sink to async. However, it will get fixed in next revision. Thanks for pointing it out!

2 and 3).  I actually thought about making HttpClient static but this will limit usability in current application design. As you can see this sink is mixture of parallel and concurrent design and client can generate lot of requests at a given time so desired behavior is to avoid blockage on client side. I am also not positive if static client will be thread safe or not. Please see https://msdn.microsoft.com/en-us/library/system.net.http.httpclient(v=vs.118).aspx#Anchor_6

In my opinion, socket waiting is not a big issue. Usually system will recycle them if needed. I believe system establish pool of connections, after all it’s all about thread pool, connection pool (db etc) and resource pool to minimize recreation cost 😊

I'm all up and hearing ...

**Comment by Morcatko** [comment]
Hm. If I google for HttpClient I get slightly different page - https://msdn.microsoft.com/en-us/library/system.net.http.httpclient(v=vs.110)
They say specifically that PostAsync is thread safe
> The following methods are thread safe:
> ...
> PostAsync

I found another MS page -  or https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client#create-and-initialize-httpclient
> HttpClient is intended to be instantiated once and reused throughout the life of an application. The following conditions can result in SocketException errors:
> - Creating a new HttpClient instance per request.
> - Server under heavy load.

Anyway I will fork your repo and try to do the changes

**Comment by saleem-mirza** [comment]
Well, this is very interesting. I'm looking forward for pull request along with test results.

**Comment by saleem-mirza** [comment]
I see that in this sink, HttpClient has it's own set of headers which can't be shared. These headers changes from request to request. 
If I create static instance of HttpClient, it can lead to potential data race condition especially when multiple threads are calling this method unless I put some kind of thread guard. Locking mechanism will essentially kill purpose of async/await. 

It's classic **Between the Devil and the deep blue sea**
